### DISCUSSION TITLE
How can I split PRs by directory?

### DISCUSSION THREAD

**Discussion by daniel-b2c2** [renovatebot/renovate#30997] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I have a 'config-as-code'/'gitops' project with the following layout:

```
environments/dev/...
environments/uat/...
environments/prod/...
```

Currently if there is an upgrade for a dependency that appears in all environments, it's offered as a single PR:

```
environments/dev/vulnerability.yml
[Upgrade from 1.0.0 -> 2.0.0]
environments/uat/vulnerability.yml
[Upgrade from 1.0.0 -> 2.0.0]
environments/prod/vulnerability.yml
[Upgrade from 1.0.0 -> 2.0.0]

[Merge]
```

Is it possible to have renovate provide the same full coverage of dev, uat, prod, but split the same upgrade PR over each environment, such that each of the following is a separate PR:

PR 1:
```
environments/dev/vulnerability.yml
[Upgrade from 1.0.0 -> 2.0.0]

[Merge]
```

PR 2:
```
environments/uat/vulnerability.yml
[Upgrade from 1.0.0 -> 2.0.0]

[Merge]
```

PR 3:
```
environments/prod/vulnerability.yml
[Upgrade from 1.0.0 -> 2.0.0]

[Merge]
```

The usecase is that we would never want to merge dev uat and prod at the same time, we would want to merge dev first, see if there were any issues, then do the same for UAT, then finally PROD, so having separate pull requests for each environment saves us time. 

Currently we're fetching the remote branch, checking out, soft reset, and manually pushing 3 separate commits.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Use additionalBranchPrefix with parentDir. Search past discussions for more details

**Comment by daniel-b2c2** [comment]
Ah, 

I have oversimplified my example, that is the correct answer, may I re-open with a more detailed scenario?

```
environments/dev/directoryA/vulnerablePackage1.yml
environments/dev/directoryB/vulnerablePackage1.yml
environments/uat/directoryC/vulnerablePackage1.yml
environments/prod/directoryD/vulnerablePackage1.yml
somewhereElse/vulnerablePackage1.yml
```

Would ideally result in:

PR1 covering dev with 2 files
PR2 covering uat with 1 file
PR3 covering prod with 1 file
PR4 covering somewhereElse with 1 file.

I can't use basedir here as I would open 2 PRs for directoryA,B which relate to the same vulnerability within the dev environment, the screenshot below shows how 1 vulnerability in multiple files within the same environment shows up as separate PRs

<img width="242" alt="image" src="https://github.com/user-attachments/assets/ddb779b5-a886-4eea-9f8d-2a96902a0a97">

I can't use parentDir here either, as the vulnerable package has another level between it.

```
  "packageRules": [
    {
      "matchFileNames": [
        "environments/dev/**",
        "environments/prod/**",
        "environments/shared/**",
        "environments/uat/**"
      ],
      "additionalBranchPrefix": "{{baseDir}}-"
    }
  ]
```

**Reply by daniel-b2c2** [reply]
I've just realised I can do this instead:

```"packageRules": [
    {
      "matchFileNames": [
        "environments/dev/**",
      ],
      "additionalBranchPrefix": "dev"
    }
  ]
```

and repeat for each environment; I was blinded by the {basedir} variable; I don't have to use variables at all!

thank you 🦆
